array-manipulation-3/src/zip.js

```diff
@@ -0,0 +1,14 @@
+/* exported zip */
+function zip(first, second) {
+  const newArray = [];
+  for (let i = 0; i < first.length; i++) {
+    if (!first[i] || !second[i]) {
+      break;
+    }
```

Comment:
What if the values in the array are falsy? You can actually decide which array is shorter before the loop starts.

```js
const length = Math.min(first.length, second.length);
```